Frontend/VIAProMa/Assets/AnchoringMenu.cs

```diff
@@ -0,0 +1,250 @@
+using System;
+using UnityEngine;
+using i5.VIAProMa.UI;
+using Microsoft.MixedReality.Toolkit.UI;
+using TMPro;
+using i5.Toolkit.Core.Utilities;
+using Photon.Pun;
+using i5.VIAProMa.Multiplayer;
+using i5.VIAProMa.UI.MainMenuCube;
+using i5.VIAProMa.Anchoring;
+
+/// <summary>
+/// Controls the menu which allows a user to enable and disable the anchoring system as well as the corresponding settings
+/// </summary>
+public class AnchoringMenu : MonoBehaviour, IWindow
+{
+    private bool windowEnabled = true;
+
+    /// <summary>
+    /// UI Elements
+    /// </summary
+    [SerializeField] private Interactable useAnchoringCheckbox;
+    [SerializeField] private Interactable moveAnchorAloneCheckbox;
+
+    [SerializeField] private GameObject extendedSettings;
+    [SerializeField] private SpriteRenderer lockedIconSpriteRenderer;
+    [SerializeField] private TextMeshPro lockButtonText;
+
+    [SerializeField] private Sprite unlockedIcon;
+    [SerializeField] private Sprite lockedIcon;
+
+    /// <summary>
+    /// The anchor in the scene as well as the 3D anchor model as a child and the corresponding manager
+    /// </summary>
+    private GameObject anchorParent;
+    private GameObject anchorObject;
+    private AnchorManager anchorManager;
+
+
+    public void Awake()
+    {
+        anchorParent = this.transform.parent.parent.parent.gameObject;
+        anchorObject = anchorParent.transform.Find("AnchorObject").gameObject;
+        anchorManager = this.transform.parent.parent.gameObject.GetComponentInChildren<AnchorManager>();
```

Comment:
I would recommend re-organizing how the menu finds the anchor manager and the anchor object itself since it now depends on exact names and object hierarchies, so it is easy to accidentaly breaking this.

Since you made the anchor manager a singleton, you should be able to directly get a reference to it by calling `AnchorManager.Instance`.
The anchor manager should administer the anchor object and so it always has a reference to it. If you create a public readonly property `public GameObject AnchorObject {get => anchorObject; }` in the AnchorManager, other scripts like the menu can access it easily by `AnchorManager.Instance.AnchorObject`.